### PR TITLE
Update devtools-extensions build script to reflect changes in local b…

### LINKED ISSUE #20997: Bug: Can't build react-devtools-extensions
I've just cloned the repo, ran `yarn`, and then `yarn build-for-devtools` as explained in the [README](https://github.com/facebook/react/blob/master/packages/react-devtools-extensions/README.md) but when trying to run `yarn build:firefox:dev` it can't find the packages I just built.

image

I think I'm following all the instructions here.

### PR DESCRIPTION
**Pull request by hecerinc**
## Summary

Updates the `react-devtools-extensions` build script to reflect changes in the local build directory structure. Specifically, the current build code assumes (in the webpack conf) the existence of `<repo_root>/build/node_modules` with either a local build or pulled from remote builds.

The changes made in #20581 changed the build output folder to `build2/oss-experimental`. Code was added in that PR to copy from `build2/oss-experimental` to `build/node_modules`, presumably to support this script, however the local build scenario was probably not considered.

This PR adds a check that validates that the build artifacts exist in `build/node_modules` and copies the `build2/oss-experimental` folder to `build/node_modules`.

I also added more refined logic to choose the correct Firefox profile based on whether the user set the `WEB_EXT_FIREFOX` env variable.

Fixes #20997

## Test Plan

image
image

### CODE REVIEW

**Comment by hecerinc** [comment]
@bvaughn here's the PR to fix the local build scenario. I wonder if I could also get a review from @acdlite  seeing as he made the changes for the other PR I reference in the description. 😄  Thanks! 🚀

File: packages/react-devtools-extensions/build.js
Comment (bvaughn) [inline packages/react-devtools-extensions/build.js:42]: Nice addition. Thanks!

File: packages/react-devtools-extensions/firefox/test.js
Comment (bvaughn) [inline packages/react-devtools-extensions/firefox/test.js]: nit I generally like to avoid abbreviations in variable names like "ff". Totally not a big deal given the context of this change, but for consistency elsewhere I might rename this one before merging.

File: packages/react-devtools-extensions/firefox/test.js
Comment (bvaughn) [inline packages/react-devtools-extensions/firefox/test.js]: nit This comment should be moved into the switch above. (I'll do this.)

**Review by bvaughn — APPROVED** [review]
Thanks 😃 The changes to the build script and the Firefox testing profile stuff look like nice improvements.

**Pull request merged**
